### ISSUE TITLE
Menu bar, assistant, etc font super small

### ISSUE THREAD

**Issue by jussikiova** [thonny/thonny#1113]
I ran into a bit of an issue with Thonny 3.2.7 on Linux Mint 19.3. The menu bar and assistant fonts are really small and won't react to neither options/theme & font or options/general font scaling mode settings. Only setting the fonts react to is options/general UI scaling factor but that scales everything else as well. I've been using version 2.1.16 and it didn't have this problem

![Thonny_small_font](https://user-images.githubusercontent.com/62099764/76515130-bc2ea400-6461-11ea-8b07-20e064d5ed96.png)

**Comment by jiri-one** [comment]
I can confirm this issue. I have absolutely same problem on Arch Linux and on Fedora 31 too. Version od Thonny is 3.2.7
![76159839-a731d800-6124-11ea-93c6-972affdd62f7](https://user-images.githubusercontent.com/32774109/77143404-83a35180-6a83-11ea-826d-202c844b41e3.png)

**Comment by aivarannamaa** [comment]
sorry for late reply. Please check whether tweaking "Tools => Options => General => Font scaling mode" helps!

**Comment by aivarannamaa** [comment]
Sorry, I now noticed that you have already tried this. I'll look into this.

**Comment by jiri-one** [comment]
On Windows is it OK. Just in Linux. Thank for your answer.
![image](https://user-images.githubusercontent.com/32774109/77147457-90c53e00-6a8d-11ea-9075-1eb23c9bb38b.png)

**Comment by Maik37** [comment]
Ich habe das selbe Problem. Ich verwende ebenfalls Linux

**Comment by KTP95** [comment]
Same problem with vers 3.2.7, kubuntu 20.04 beta, and display 4k.
But... there are a fix, with UI Theme Raspberry Dark (or light) is acceptable sizing,
ps. Need restart Thonny.

But i generally with any UI Theme find another issue:

Color Patten of "MENU TOOLTIPs" is terrible, why is dark-background as all?

![Image of Tooltips](https://i.imgur.com/vibWQFF.png)

**Comment by Maik37** [comment]
For me this didn't work. Even if I'm restarting, it's not changing.

**Comment by aivarannamaa** [comment]
Have you installed Thonny with system package manager (in this case it uses system-provided Tk) or Thonny+Python binary bundle (which also contains Tk)?

**Comment by Maik37** [comment]
It was already installed in my Raspberry Pi 4.

**Comment by Maik37** [comment]
It was already installed on my raspberry pi 4

**Comment by speedy-10** [comment]
Have the same problem as jussikovia with Ubuntu 20.04.1 (Gnome-Desktop) using 3840 x 2160 resolution.

**Comment by K9Kraken** [comment]
Using Thonny on Manjaro and the font is way too small.
Thonny 3.3.3
![manjaro_thonny_tiny_font](https://user-images.githubusercontent.com/37868357/106408508-da93ea00-63fb-11eb-82bb-7594ad0b40a4.png)

**Comment by zyuri** [comment]
Any updates on this? 
I have the same with Thonny 3.3.13 and KUbuntu 21.04...

**Comment by speedy-10** [comment]
No Problems with Ubuntu 20.04, Thonny 3.3.13., UI Theme "Enhaced clam". Regards

**Comment by zyuri** [comment]
When chaning the thonny script to start thonny so that it uses my system's  python 3.9 it instead of the venv works...
#!/bin/bash
# next two lines are only needed when you have an issue with BACKGROUND color
#xrdb -load /dev/null
#xrdb -query

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"

#export LD_LIBRARY_PATH="$DIR/../lib:$LD_LIBRARY_PATH"
export SSL_CERT_DIR=$(readlink -m "$DIR/../lib/python3.7/site-packages/certifi")
export SSL_CERT_FILE="$SSL_CERT_DIR/cacert.pem"

unset PYTHONHOME
unset PYTHONPATH
unset PYTHONSTARTUP
unset PYTHONINSPECT
#unset TCL_LIBRARY
#unset TK_LIBRARY

#"$DIR/python3.7" -E -m thonny "$@"
python3 -E -m thonny "$@"

**Comment by j0fiN** [comment]
![image](https://user-images.githubusercontent.com/55355647/136139708-d8598597-eb38-40af-8e0a-133ddfb69575.png)

I have the same problem with Windows 10.
version: 3.3.13

How do I resolve this?

**Comment by gatonero** [comment]
Same here in Manjaro
![thonny](https://user-images.githubusercontent.com/8248255/160935138-65cea5e6-ca61-4a1c-8b84-487b387bab65.png)

**Comment by yaconsult** [comment]
Can confirm tiny, unusable menu fonts on fresh install of fedora 36 with thonny 3.3.14.  The increased and decreased font menu only affect the program windows, not the menubar.  I tried different themes as suggested above but none of them increased the size of the menu text.  The display is 1920x1200 on X11 using the latest Nvidia drivers from rpmfusion.
![image](https://user-images.githubusercontent.com/1074939/172084620-0427a0fb-ac72-4d0f-b483-aac47952ea13.png)

**Comment by ErsatzMoco** [comment]
Same here, but there's a workaround: Set "Options/General/Font scaling mode" to "extra". At least in my case that restores a reasonable menu font size.

If you can't see this option item, jump to "Tools/Open Thonny data folder" and edit "configuration.ini". Find "scaling" and set it to a lower value. In the configuration file there's also "font_scaling_mode", you can set it to "extra" here as well (omit quotes, see current .ini entry).

Looks like upscaling the UI downscales the menu font in Linux or something like that, at least with font_scaling_mode = default.

**Comment by yaconsult** [comment]
Correct.  I found another issue here complaining that the fonts were too large and based on the solution, I edited configuration.ini and changed "scaling = default" to "scaling = 1.0" and this fixed the problem of UI fonts being tiny.  The picture in my post above shows what thonny looked like before the edit.  This is what it looked like after:
![image](https://user-images.githubusercontent.com/1074939/175380479-7214f8e3-5945-4a7f-8437-1ac7333170b1.png)

**Comment by Zaptrouiyerxi** [comment]
Nothing from above works in any combination to fix the only remaining unworkable detail for me: the size of checkboxes 

![image](https://github.com/thonny/thonny/assets/35660751/3a387bbb-8150-4854-80f5-9ef41818f912)

**Comment by MartinRGB** [comment]
In `Ubuntu 22.04.3 LTS x86_64`

If you use `sudo snap install thonny`, you will encounter this problem, if you use `sudo apt install thonny`, you will not, and thonny versions are different.

**Comment by alxxthegeek** [comment]
MartinRGB no. I installed via sudo apt install thonny and have the font problem on Ubuntu mate 23.10 with menu's only visible in dark mode and small font size for menu. That's with thonny v4.0.1

**Comment by AtomicRobotMan0101** [comment]
**A Fix** 

**System:** Mint 22.1 Xia, X11, Cinnamon, Nvidia, 4608x2430 @60Hz and 275% screen scaling (I use my TV). 

Version is 4.1.7 via the `bash` script method, `apt` version and `flatpak` version were uninstalled (in frustration) and remain untested, but I don't see why this wouldn't work.

Edit `.config/Thonnyconfiguration.ini` and mod the line (about line 9) to `scaling = 10.0`

restart Thonny and fonts will be nicely sized again.

**Note:** tried other scaling factors (e.g. 6.0, 8.0, etc) and the restart of Thonny simply overrode them. Others above 10 seem OK.  I ended up on 11. I'm not bothered by the small icon size, for they are in the way anyway.

Lastly, thank you devs for Thonny. Slava Ukraini!

<img width="2446" height="1339" alt="Image" src="https://github.com/user-attachments/assets/1050660e-c2ce-4447-99e9-c5650ce2aa51" />

**Comment by YorkEarwaker** [comment]
Dear @jussikiova ,

Had similar problem on Ubuntu LTS 24.04.3 Linux on a Dell laptop. Like @AtomicRobotMan0101 and @yaconsult the general.scaling property in the configuration.ini file seemed to fix the issue.

In this instance also added it to the configuration.ini file. This item is also available via the Thonny menu. However if the menu items are too small to see without magnification then hard to change value via the menu.

Opened configuration.ini file in Text Editor in Ubuntu added item to config and saved. Then opened Thonny and the issue was resolved. Location of the config file on Linux for the Ubuntu snap install Thonny instance is described in the link below.

Please find some notes on same at the following link. This also discusses some other property values in the config file. Thonny IDE [WS](https://github.com/YorkEarwaker/Automation/tree/main/ide/thy#issue-1) ,

<img width="2968" height="1115" alt="Image" src="https://github.com/user-attachments/assets/ef0bdfcc-1800-44fb-9f0a-396333709f13" />

<img width="1556" height="1195" alt="Image" src="https://github.com/user-attachments/assets/e6e2724a-1242-4d17-b73a-41435e615990" />


Please consider setting the property general.scaling = 2.0 as default in the configuration.ini file for the Ubuntu snap package of Thonny .

BTW. Thonny is an excellent MicroPython IDE for introduction to micro electronics and MCU's. Good choice. Only started to use the Thonny IDE fairly recently and it has been a rewarding experience.

Yours,
York
